src/main/java/org/folio/services/domainevent/CommonDomainEventPublisher.java

```diff
@@ -179,6 +180,10 @@ private Future<Void> publish(String key, Object value) {
 
     KafkaProducer<String, String> producer = getOrCreateProducer();
 
+    if (okapiHeaders.get(XOkapiHeaders.TOKEN.toLowerCase()) != null) {
```

Comment:
Does this mean shadow instances are updated with the same token that updated the shared instance? If so, what if the user that updated the shared instance does not have an affiliation to a shadow instance's tenant?

Reply:
yes correct, this token will be received by this handler. It will work fine because the handler using this token performs http requests to central tenant. Just double-checked this case on the rancher env that it works as expected.

Reply:
can the token perform http requests to member tenants if the user of the token does not have an affiliation?

Reply:
nevermind! I was thinking we use the token to update shadow instance. But this is inventory-storage, a token is only used to get the latest central tenant instance.